File: archetype-java/src/main/resources/archetype-resources/src/main/java/AppConfig.java
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package};

import com.netflix.config.DynamicPropertyFactory;
import com.netflix.config.DynamicStringProperty;
import org.springframework.context.annotation.*;
import org.springframework.context.annotation.ComponentScan.Filter;
import org.springframework.stereotype.Controller;
import thirtytwo.degrees.halfpipe.config.DefaultAppConfg;
import thirtytwo.degrees.halfpipe.mgmt.resources.GCResource;

import javax.inject.Named;

@Configuration
@ComponentScan(basePackageClasses = {AppConfig.class, GCResource.class}, excludeFilters = {
    @Filter(Controller.class),
    @Filter(Configuration.class)
})
@ImportResource("classpath:META-INF/spring/applicationContext-security.xml")
@Import(DefaultAppConfg.class)
public class AppConfig {

    @Bean @Named("helloText")
    public DynamicStringProperty helloText() {
        return DynamicPropertyFactory.getInstance().getStringProperty("hello.text", "Hello default text");
    }

    @Bean
    public GCResource garbageCollectionTask() {
        return new GCResource();
    }

}

File: archetype-scala/src/main/resources/archetype-resources/src/main/scala/AppConfig.scala
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}

import org.springframework.context.annotation._
import com.netflix.config.scala.DynamicProperties
import javax.inject.Named
import org.springframework.context.annotation.ComponentScan.Filter
import org.springframework.stereotype.Controller
import thirtytwo.degrees.halfpipe.DefaultScalaAppConfig

@Configuration
@ComponentScan (basePackageClasses = Array (classOf[AppConfig]),
  excludeFilters = Array(new Filter (Array (classOf[Controller], classOf[Configuration]))))
@Import(Array(classOf[DefaultScalaAppConfig], classOf[SecurityConfig]))
class AppConfig extends DynamicProperties {

  @Bean @Named("helloText")
  def helloText() = dynamicStringProperty("hello.text", "Hello default")

}